ocaml/libs/log/debug.ml

```diff
@@ -310,6 +310,10 @@ module type DEBUG = sig
   val log_backtrace : unit -> unit
 
   val log_and_ignore_exn : (unit -> unit) -> unit
+
+  val log_and_ignore_exn1 : ('a -> unit) -> 'a -> unit
```

Comment:
Are these needed? The first log_and_ignore_exn can be used with functions of any number or arguments:
```ocaml
D.log_and_ignore_exn2 (fun () -> release resource acquired) ;
```

Reply:
That would require allocating a closure, the advantage of having these specialised 1 and 2 argument functions is that closure allocation can be avoided. Since these functions would be used in fast-paths having the optimized version is worhtwhile.

Reply:
I'm going to build a switch with 'flambda' to test and compare, if there is a significant benefit in general we might consider testing and turning on flambda by default (it'll avoid having to do lots of these small optimizations by hand).
(Compilation times might increase, but only on Koji)

Reply:
I reverted `f590dc` and added '-O3' and '-unbox-closures' to both the benchmark's and xapi-internal's ocamlopt_flags and looked at the number of allocations.
Without modifying xapi_internal's flags I got 10 'minor-allocated', and when modifying it I got back ~5, which is what I get when *not* reverting the commit.

So it looks like 'flambda' can perform this optimization already on its own and it is worth-while to further investigate applying flambda to XAPI in general:

flambda -O3 -unbox-closures just on the benchmark:
```
╭────────────────────────────┬───────────────────────────┬───────────────────────────┬───────────────────────────╮
│name                        │  major-allocated          │  minor-allocated          │  monotonic-clock          │
├────────────────────────────┼───────────────────────────┼───────────────────────────┼───────────────────────────┤
│  Mutex/Mutex.execute       │             0.0000 mjw/run│             2.9412 mnw/run│            488.3654 ns/run│
│  Mutex/Mutex.lock/unlock   │             0.0000 mjw/run│             2.8302 mnw/run│            489.5659 ns/run│
│  Mutex/NamedMutex.execute  │             0.0000 mjw/run│            10.9412 mnw/run│           3268.9580 ns/run│
╰────────────────────────────┴───────────────────────────┴───────────────────────────┴───────────────────────────╯
```

flambda -O3 -unbox-closures also on xapi-internal:
```
╭────────────────────────────┬───────────────────────────┬───────────────────────────┬───────────────────────────╮
│name                        │  major-allocated          │  minor-allocated          │  monotonic-clock          │
├────────────────────────────┼───────────────────────────┼───────────────────────────┼───────────────────────────┤
│  Mutex/Mutex.execute       │             0.0000 mjw/run│             7.0612 mnw/run│            689.6918 ns/run│
│  Mutex/Mutex.lock/unlock   │             0.0000 mjw/run│             2.8302 mnw/run│            519.8271 ns/run│
│  Mutex/NamedMutex.execute  │             0.0000 mjw/run│             4.6316 mnw/run│           2682.6565 ns/run│
╰────────────────────────────┴───────────────────────────┴───────────────────────────┴───────────────────────────╯```